app/views/mods_panel.py

```diff
@@ -1538,6 +1574,32 @@ def append_new_item(self, uuid: str) -> None:
         item.setData(Qt.ItemDataRole.UserRole, data)
         self.addItem(item)
 
+    def get_all_loaded_mod_list_items(self) -> list[ModListItemInner]:
+        # This gets all modlist items as ModListItemInner.
+        # Mods that have not been loaded
+        # or lazy loaded will not be returned.
+        mod_list_items = []
+        for index in range(self.count()):
+            item = self.item(index)
+            widget = self.itemWidget(item)
+            if isinstance(widget, ModListItemInner):
+                mod_list_items.append(widget)
+        return mod_list_items
+
+    def get_all_loaded_and_toggled_mod_list_items(self) -> list[ModListItemInner]:
+        # This gets all modlist items that have their warnings toggled as ModListItemInner
+        # Mods that have not been loaded
+        # or lazy loaded will not be returned.
+        mod_list_items = []
+        for index in range(self.count()):
+            item = self.item(index)
+            widget = self.itemWidget(item)
+            if isinstance(widget, ModListItemInner):
+                packageid = self.metadata_manager.internal_local_metadata[widget.uuid].get('packageid')
+                if packageid in self.ignore_warning_list:
+                    mod_list_items.append(widget)
+        return mod_list_items
```

Comment:
Because this only gets non-lazy items, if the user refreshes and then tries to reset toggles, lazy items are not reset, even after loading them.

Reply:
To solve this, you may want to handle the toggle state in the QListWidgetItem's data. In relation to this, if you do this, because of the increasing complexity of the data, you may also want to change the data to be a class instead of a dict/map.